### PR TITLE
Implement zone creation flows

### LINKED ISSUE #101: Implement zone creation flows
Implement the zone creation flows for all zone shapes (Box, Cylinder, Point, Route) when triggered from the Zone Management GUI.

## Technical Notes
- Triggered by clicking shape button (Box/Cylinder/Point/Route) in Create New Zone section
- Closes GUI and enters zone creation mode with visual/audio feedback
- Shape-specific placement:
  - **Box**: Two clicks define opposite corners
  - **Cylinder**: Click sets center, then GUI prompts for radius and height
  - **Point**: Single click defines position
  - **Route**: Multiple clicks add waypoints, sneak+right-click finalizes
- After placement, configuration GUI opens to set zone name and zone type
- Creates zone with appropriate bounds/waypoints
- Exits creation mode and returns to Zone Management GUI after creation

### Rendering Reuse
**Dependency:** Requires rendering infrastructure from #25

Reuse zone rendering utilities (`ZoneRenderUtil.renderBox()`, etc.) created in #25 for preview visualization during zone creation:
- **Data source**: In-progress zone being created (not synced zones)
- **Partial state rendering**:
  - Box: First corner + cursor position for dynamic second corner wireframe
  - Cylinder: Center point + growing radius as cursor moves
  - Route: Connected waypoints with cursor position as next potential point
- **Visual distinction**: Different color for preview (e.g., yellow/orange) vs. actual zones (white/cyan)

## Related Issues
- Blocked by #25 (requires zone rendering utilities)

## Testing
- Test each shape button triggers creation mode correctly
- Verify Box: two corners define bounds correctly, preview renders during placement
- Verify Cylinder: center + dimensions create correct cylindrical bounds, preview renders
- Verify Point: single click creates zone at position, marker renders
- Verify Route: multiple waypoints sequence correctly, finalization works, path renders during creation
- Test configuration GUI appears and saves name/type properly for all shapes
- Cancel during any placement flow, verify clean exit and preview disappears
- Test edge cases (extreme coordinates, minimum/maximum values)

## Acceptance Criteria
- [ ] All four shape buttons trigger creation mode
- [ ] Box zones created with two-corner placement and dynamic preview rendering
- [ ] Cylinder zones created with center + dimension input and preview rendering
- [ ] Point zones created with single-click placement and marker rendering
- [ ] Route zones created with multi-waypoint + finalization and path rendering
- [ ] Preview rendering uses shared utilities from #25
- [ ] Configuration GUI works for all zone types
- [ ] Visual/audio feedback active during creation mode
- [ ] Creation mode exits cleanly after creation or cancellation

### LINKED ISSUE #102: Implement visual and audio feedback for zone creation mode
Implement visual and audio feedback for when the Ledger is in zone creation mode.

## Technical Notes
- Enchantment glimmer effect on the ledger item while in creation mode
- Periodic tinkling sound effect while in creation mode
- Particle effects when hovering over valid click targets (blocks)
- Visual indication of placed waypoints/corners during multi-step creation
- Feedback should be client-side only to avoid network overhead
- Sound volume should be subtle and not annoying during extended use

## Testing
- Enter creation mode, verify enchantment glimmer appears on ledger
- Listen for periodic tinkling sound, verify it plays at reasonable intervals
- Hover over blocks, verify particle effects appear
- Place first corner/waypoint, verify visual marker appears
- Exit creation mode, verify all effects stop
- Test in multiplayer, verify other players see appropriate effects

## Acceptance Criteria
- [ ] Enchantment glimmer renders on ledger during creation mode
- [ ] Periodic tinkling sound plays while in creation mode
- [ ] Particle effects appear on valid click targets
- [ ] Visual markers show placed waypoints/corners during multi-step creation
- [ ] All effects stop when exiting creation mode
- [ ] Effects render properly in multiplayer
- [ ] Performance remains acceptable with effects active

### PR DESCRIPTION
**Pull request by nivthefox**
# Implement zone creation flows
Completes zone creation system by implementing all four zone shape workflows (Box, Cylinder, Point, Route) with visual feedback and dynamic previews.

## Changes
- Add interactive box zone creation with two-corner placement
- Add interactive cylinder zone creation with center + dimension workflow
- Add interactive point zone creation with single-click placement
- Add interactive route zone creation with multi-waypoint + sneak-finalize workflow
- Implement real-time wireframe preview rendering for all zone shapes during creation
- Add visual effects (particles + sound) during zone creation mode
- Add zone configuration GUI for naming and type selection after placement
- Unify zone rendering through BoundaryRenderer for both creation previews and F3+V overlay

## Related Issues
Resolves #101
Resolves #102 

## Implementation Notes
- Reused existing zone rendering utilities from #25 for preview visualization
- Route creation uses sneak+right-click to finalize (matches issue spec)
- Creation mode auto-cancels if player switches away from ledger
- Zone creation state persists on client via SetZoneCreationMode packet
- All zone shapes support preview rendering with distinct colors (yellow/orange for preview vs white/cyan for actual zones)

## Breaking Changes
None

### CODE REVIEW

File: src/main/java/org/sosly/villagetale/server/ServerEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+@Mod.EventBusSubscriber(modid = VillageTale.MOD_ID)
+public class ServerEventHandler {
+
+    @SubscribeEvent
+    public static void onPlayerLogout(PlayerLoggedOutEvent event) {
+        ZoneCreationHelper.getInstance().removePlayer(event.getEntity().getUUID());
+    }
+
+    @SubscribeEvent
+    public static void onPlayerInteract(PlayerInteractEvent event) {
```
[the remark this answers is not in the document]
Comment (nivthefox) [reply src/main/java/org/sosly/villagetale/server/ServerEventHandler.java:20]: This is desired behavior.

File: src/main/java/org/sosly/villagetale/client/ClientEventHandler.java [hunk 2]
```diff
@@ -1,18 +1,159 @@
...
+        if (!isActive) {
+            return;
+        }
+
+        Minecraft mc = Minecraft.getInstance();
+        if (mc.player == null) {
+            return;
+        }
+
+        playAmbientSound(mc.player);
```
[the remark this answers is not in the document]
Comment (nivthefox) [reply src/main/java/org/sosly/villagetale/client/ClientEventHandler.java:56]: This is desired behavior.
